finagle-netty4/src/main/scala/com/twitter/finagle/netty4/ssl/server/SslServerVerificationHandler.scala

```diff
@@ -24,21 +23,22 @@ private[netty4] class SslServerVerificationHandler(
   private[this] val onHandshakeComplete = Promise[Unit]()
 
   private[this] def verifySession(session: SSLSession, ctx: ChannelHandlerContext): Unit = {
```

Comment:
ditto on needing to be async

Reply:
Looks like the server side code here waits for the successful result of sessionVerifier, but we should have a test for this nonetheless.

Reply:
So the logic for the server verifier is correct? I'll try to write a test if so